plugins/modules/dcnm_inventory.py

```diff
@@ -56,13 +57,16 @@
         required: true
       auth_proto:
         description:
-        - Name of the authentication protocol to be used
+        - Name of the authentication protocol to be used.
+          For POAP configurations authentication protocol should be 'MD5'.
         choices: ['MD5', 'SHA', 'MD5_DES', 'MD5_AES', 'SHA_DES', 'SHA_AES']
         type: str
-        required: true
+        required: false
+        default: 'MD5'
       user_name:
         description:
-        - Login username to the switch
+        - Login username to the switch.
+          For POAP configurations username should be 'admin'
```

Comment:
Should we automatically set this in the code an ignore any value provided in the playbook?

Reply:
I do have a check for the username in POAP cases, if anything else is provided apart from admin, it throws an error. This is a mandatory parameter for all inventory cases. Did not want to disturb it by using default parameters for the sake of consistency.